runtime/vm/jvminit.c

```diff
@@ -8066,6 +8073,18 @@ freeClassNativeMemory(J9HookInterface** hook, UDATA eventNum, void* eventData, v
 	}
 }
 
+#if JAVA_SPEC_VERSION >= 22
+static void
+vmHookAnonClassesUnload(J9HookInterface** hook, UDATA eventNum, void* eventData, void* userData)
+{
+	J9VMAnonymousClassesUnloadEvent * unloadedEvent = (J9VMAnonymousClassesUnloadEvent *)eventData;
+	J9VMThread * vmThread = unloadedEvent->currentThread;
+	for (J9Class* j9clazz = unloadedEvent->anonymousClassesToUnload; j9clazz; j9clazz = j9clazz->gcLink) {
+		hashClassTableMHProxyPackageDelete(vmThread, j9clazz->classLoader, j9clazz->romClass);
```

Comment:
Currently it is only for `MHProxy` class. But this may extend to other classes in the future. I suggest not to include `MHProxy` in the function name.